src/lib/y2storage/boot_requirements_strategies/base.rb

```diff
@@ -201,6 +212,25 @@ def unknown_boot_disk_error
         error_message = _("Boot requirements cannot be determined because there is no '/' mount point")
         SetupError.new(message: error_message)
       end
+
+      # Whether the filesystem containing /boot is readable by grub
+      #
+      # We might need to check if the filesystem is actually supported by grub
+      # but currently all storage-ng supported filesystems are.
+      #
+      # @return [Boolean] true if the filesystem where /boot resides is going to
+      #   be readable by grub
+      def grub_can_read_boot?
```

Comment:
No sure about this level of indirection. Why do we need `#grub_can_read_boot?` if it only calls `#encrypted_for_grub?`. Both are private and none of them is redefined in a derivate class. So I don't see the point right now.

Moreover, directly using a method that includes "encrypted" in its name would make the part above more understandable. I mean this:

```ruby
if !encrypted_for_grub?
  error_message = _("....Only LUKS1 encryption is supported...")
  res << SetupError.new(message: error_message)
end
```

Reply:
fair enough